src/shared/components/views/OrganismDataView.tsx

```diff
@@ -0,0 +1,38 @@
+import { FC } from 'react';
+import { Link } from 'react-router-dom';
+
+import { getEntryPath } from '../../../app/config/urls';
+
+import { Namespace } from '../../types/namespaces';
+import { OrganismData } from '../../../uniprotkb/adapters/namesAndTaxonomyConverter';
+
+type Props = {
+  organism: OrganismData;
+  displayOnlyID?: boolean;
+};
+
+export const OrganismDataView: FC<Props> = ({ organism, displayOnlyID }) => {
+  if (!organism.taxonId) {
+    // eslint-disable-next-line no-console
+    console.warn("No taxon ID, this shouldn't happen", organism);
+    return null;
+  }
+  const { scientificName, commonName, taxonId } = organism;
+  let name;
+  if (scientificName) {
+    name = scientificName;
+    if (commonName) {
+      name += ` (${commonName})`;
+    }
+  } else if (commonName) {
+    name = commonName;
+  }
```

Comment:
To be paranoid here the case that neither `scientificName` or `commonName` exists isn't handled.